src/sia.c

```diff
@@ -1,31 +1,34 @@
 #include "sia.h"
 
 #include <cx.h>
+#include <ledger_assert.h>
+#include <lib_standard_app/crypto_helpers.h>
 #include <os.h>
+#include <os_seed.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
 
 #include "blake2b.h"
 
-void deriveSiaKeypair(uint32_t index, cx_ecfp_private_key_t *privateKey, cx_ecfp_public_key_t *publicKey) {
-    uint8_t keySeed[32];
-    cx_ecfp_private_key_t pk;
+static void siaSetPath(uint32_t index, uint32_t path[static 5]) {
+    path[0] = 44 | 0x80000000;
+    path[1] = 93 | 0x80000000;
+    path[2] = index | 0x80000000;
+    path[3] = 0x80000000;
+    path[4] = 0x80000000;
+}
 
-    // bip32 path for 44'/93'/n'/0'/0'
-    uint32_t bip32Path[] = {44 | 0x80000000, 93 | 0x80000000, index | 0x80000000, 0x80000000, 0x80000000};
-    os_perso_derive_node_bip32_seed_key(HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32Path, 5, keySeed, NULL, NULL, 0);
+void deriveSiaPublicKey(uint32_t index, uint8_t raw_pubkey[static 64]) {
+    uint32_t bip32Path[5];
+    siaSetPath(index, bip32Path);
 
-    cx_ecfp_init_private_key(CX_CURVE_Ed25519, keySeed, sizeof(keySeed), &pk);
-    if (publicKey) {
-        cx_ecfp_init_public_key(CX_CURVE_Ed25519, NULL, 0, publicKey);
-        cx_ecfp_generate_pair(CX_CURVE_Ed25519, publicKey, &pk, 1);
-    }
-    if (privateKey) {
-        *privateKey = pk;
+    if (raw_pubkey) {
+        if (bip32_derive_with_seed_get_pubkey_256(HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32Path, 5, raw_pubkey, NULL, CX_SHA512, NULL, 0) != CX_OK) {
+            ASSERT_DISPLAY_MESSAGE("get pubkey failed");
+            return;
```

Comment:
This is not how it should be used!
You should only use `LEDGER_ASSERT`().
Correct usage would look like:

```
ret = bip32_derive_with_seed_get_pubkey_256(HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32Path, 5, raw_pubkey, NULL, CX_SHA512, NULL, 0);
LEDGER_ASSERT(ret == CX_OK, "get pubkey failed"); // doesn't return in case of test failure
```